packages/react-native/React/Views/RCTConvert+Transform.m

```diff
@@ -141,7 +148,50 @@ + (CATransform3D)CATransform3D:(id)json
       RCTLogInfo(@"Unsupported transform type for a CATransform3D: %@.", property);
     }
   }
+  
+  transform = CATransform3DTranslate(transform, -translateX, -translateY, -translateZ);
   return transform;
 }
 
++ (NSArray *)getTranslateForTransformOrigin:(CGFloat)viewWidth viewHeight:(CGFloat)viewHeight transformOrigin
+:(NSString*)transformOrigin {
+  if (transformOrigin.length == 0 || (viewWidth == 0 && viewHeight == 0)) {
+    return @[@(0.0), @(0.0), @(0.0)];
```

Comment:
Can we return `nil` in this case, and use that to avoid a call to `CATransform3DTranslate`?